src/modules/dotnet.rs

```diff
@@ -0,0 +1,308 @@
+use ansi_term::Color;
+use std::ffi::OsStr;
+use std::iter::Iterator;
+use std::ops::Deref;
+use std::path::Path;
+use std::process::{Command, Stdio};
+use std::str;
+
+use super::{Context, Module};
+
+type JValue = serde_json::Value;
+
+const DOTNET_SYMBOL: &str = "•NET ";
+const GLOBAL_JSON_FILE: &str = "global.json";
+const PROJECT_JSON_FILE: &str = "project.json";
+
+/// A module which shows the latest (or pinned) version of the dotnet SDK
+///
+/// Will display if any of the following files are present in
+/// the current directory:
+/// global.json, project.json, *.sln, *.csproj, *.fsproj, *.xproj
+pub fn module<'a>(context: &'a Context) -> Option<Module<'a>> {
+    let dotnet_files = get_local_dotnet_files(context).ok()?;
+    if dotnet_files.is_empty() {
+        return None;
+    }
+
+    let mut module = context.new_module("dotnet");
+
+    // Internally, this module uses its own mechanism for version detection.
+    // Typically it is twice as fast as running `dotnet --version`.
+    let enable_heuristic = module.config_value_bool("heuristic").unwrap_or(true);
+    let version = if enable_heuristic {
+        let repo_root = context
+            .get_repo()
+            .ok()
+            .and_then(|r| r.root.as_ref().map(|buf| buf.as_path()));
```

Comment:
```suggestion
            .and_then(|r| r.root.as_ref().map(PathBuf::as_path))
```

I'm not 100% certain but I believe you can do this.

Reply:
I will try! 😄

Reply:
Yep, this worked. I've made the change.